src/test/mochitest/head.js

```diff
@@ -343,13 +343,13 @@ function isPaused(dbg) {
  * @static
  */
 async function waitForPaused(dbg) {
-  // We want to make sure that we get both a real paused event and
-  // that the state is fully populated. The client may do some more
-  // work (call other client methods) before populating the state.
-  let loading = waitForDispatch(dbg, "LOAD_OBJECT_PROPERTIES");
-  await waitForThreadEvents(dbg, "paused");
-  await waitForState(dbg, state => isPaused(dbg));
-  await loading;
+  const { getSelectedScope, hasLoadingObjects } = dbg.selectors;
+
+  return waitForState(
```

Comment:
much easier to read. One question, is `isPaused` equivalent to the `waitForThreadEvents`?

Reply:
isPaused, is not the same as `waitForThreadEvents`, but I intentionally dropped `waitForThreadEvents(dbg, "paused");`

* waiting for an event means we have to start listening before we do an action otherwise we can miss it
* it is sufficient to check the pause state to see if we're paused